docs/features/deploy.md

```diff
@@ -0,0 +1,130 @@
+environment_secrets = [
+  "YOUR_COMMON_SECRET",
+]
+
+[deploy.localnet]
+command = "your localnet-specific command"  # Optional. If not specified, the generic command from [deploy] will be used.
+environment_secrets = [
+  "YOUR_LOCALNET_SECRET",
+]
+```
+
+The `command` key under each `[deploy.{network_name}]` section should contain a string that represents the deployment command for that particular network. If a `command` key is not provided in a network-specific section, the command from the general `[deploy]` section will be used.
+
+The `environment_secrets` key should contain a list of names of environment variables that should be treated as secrets. This can be defined in the general `[deploy]` section, as well as in the network-specific sections. The environment-specific secrets will be added to the general secrets during deployment.
+
+The `[algokit]` section with the `min_version` key allows you to specify the minimum version of AlgoKit that the project requires.
+
+This way, you can define common deployment logic and environment secrets in the `[deploy]` section, and provide overrides or additions for specific environments in the `[deploy.{environment_name}]` sections.
+
+## Deploying to a Specific Network
+
+The command requires a `ENVIRONMENT` argument, which specifies the network environment to which the smart contracts will be deployed. Please note, the `environment` argument is case-sensitive.
+
+Example:
+
+```sh
+$ algokit deploy testnet
+```
+
+This command deploys the smart contracts to the testnet.
+
+## Custom Project Directory
+
+By default, the deploy command looks for the `.algokit.toml` file in the current working directory. You can specify a custom project directory using the `--project-dir` option.
+
+Example:
+
+```sh
+$ algokit deploy testnet --project-dir="path/to/project"
+```
+
+## Custom Deploy Command
+
+You can provide a custom deploy command using the `--custom-deploy-command` option. If this option is not provided, the deploy command will be loaded from the `.algokit.toml` file.
+
+Example:
+
+```sh
+$ algokit deploy testnet --custom-deploy-command="your-custom-command"
+```
+
+## CI Mode
+
+By using the `--ci` or `--non-interactive` flag, you can skip the interactive prompt for mnemonics.
+
+This is useful in CI/CD environments where user interaction is not possible. When using this flag, you need to make sure that the mnemonics are set as environment variables.
+
+Example:
+
+```sh
+$ algokit deploy testnet --ci
+```
+
+## Example of a Full Deployment
+
+```sh
+$ algokit deploy testnet --custom-deploy-command="your-custom-command"
+```
+
+This example shows how to deploy smart contracts to the testnet using a custom deploy command and skipping the warning prompt. This also assumes that .algokit.toml file is present in the current working directory, and .env.testnet file is present in the current working directory and contains the required environment variables for deploying to TestNet environment.
```

Comment:
Is the bit about a warning prompt still relevant?

Reply:
good catch, removed the part mentioning prompt